moto/ecs/models.py

```diff
@@ -318,6 +322,42 @@ def __init__(
         self.region_name = cluster.region_name
         self._account_id = backend.account_id
         self._backend = backend
+        self.attachments = []
+
+        if task_definition.network_mode == "awsvpc":
+            if not networking_configuration:
+
+                raise InvalidParameterException(
+                    "Network Configuration must be provided when networkMode 'awsvpc' is specified."
+                )
+
+            self.network_configuration = networking_configuration
+            net_conf = networking_configuration["awsvpcConfiguration"]
+            ec2_backend = ec2_backends["123456789012"]["us-east-1"]
+
+            eni = ec2_backend.create_network_interface(
+                subnet=net_conf["subnets"][0],
+                private_ip_address=random_private_ip(),
+                group_ids=net_conf["securityGroups"],
+                description="moto ECS",
+            )
+            eni.status = "in-use"
+            eni.device_index = 0
+
+            self.attachments.append(
+                {
+                    "id": str(uuid.uuid4()),
```

Comment:
```suggestion
                    "id": str(mock_random.uuid4()),
```
The `mock_random`-object was introduced a week ago or so, and gives us the option to make random identifiers (such as uuid's) deterministic; see http://docs.getmoto.org/en/latest/docs/configuration/recorder/index.html#deterministic-identifiers
Seeding Moto only works if all our identifiers are generated by the same `Random`-object, which is what happens within the `MockRandom`-object